src/main/java/dev/sigstore/fulcio/client/FulcioValidator.java

```diff
@@ -0,0 +1,126 @@
+import java.nio.charset.StandardCharsets;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.cert.*;
+import java.security.spec.EncodedKeySpec;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Collections;
+import java.util.Date;
+import org.conscrypt.ct.CTLogInfo;
+import org.conscrypt.ct.CertificateEntry;
+import org.conscrypt.ct.VerifiedSCT;
+
+public class FulcioValidator {
+  private final CTLogInfo ctLogInfo;
+  private final TrustAnchor fulcioRoot;
+
+  public static FulcioValidator newFulcioValidator(byte[] ctfePublicKey, byte[] fulcioRoot)
+      throws InvalidKeySpecException, NoSuchAlgorithmException, CertificateException, IOException,
+          InvalidAlgorithmParameterException {
+    KeyFactory keyFactory = KeyFactory.getInstance("EC");
+    PemReader pemReader =
+        new PemReader(
+            new InputStreamReader(new ByteArrayInputStream(ctfePublicKey), StandardCharsets.UTF_8));
+    PemReader.Section section = pemReader.readNextSection();
+    if (pemReader.readNextSection() != null) {
+      throw new InvalidKeySpecException(
+          "ctfe public key must be only a single PEM encoded public key");
+    }
+    EncodedKeySpec publicKeySpec = new X509EncodedKeySpec(section.getBase64DecodedBytes());
+    PublicKey ctfePublicKeyObj = keyFactory.generatePublic(publicKeySpec);
+
+    CertificateFactory certificateFactory = CertificateFactory.getInstance("X.509");
+    X509Certificate fulcioRootObj =
+        (X509Certificate)
+            certificateFactory.generateCertificate(new ByteArrayInputStream(fulcioRoot));
+
+    TrustAnchor fulcioRootTrustAnchor = new TrustAnchor(fulcioRootObj, null);
+    // this should throw the InvalidAlgorithmException a bit earlier
+    new PKIXParameters(Collections.singleton(fulcioRootTrustAnchor));
+
+    // TODO: this ctloginfo probably needs to be appropriately filled out
+    CTLogInfo ctLogInfo =
+        new CTLogInfo(ctfePublicKeyObj, "fulcio ct log", "intentionally-nonsense");
+    return new FulcioValidator(ctLogInfo, fulcioRootTrustAnchor);
+  }
+
+  private FulcioValidator(CTLogInfo ctLogInfo, TrustAnchor fulcioRoot) {
+    this.ctLogInfo = ctLogInfo;
+    this.fulcioRoot = fulcioRoot;
+  }
+
+  public void validateSct(SigningCertificate sc)
+      throws CertificateEncodingException, FulcioValidationException {
+    if (!(sc.getLeafCertificate() instanceof X509Certificate)) {
+      throw new RuntimeException(
+          "Encountered non X509 Certificate when validating SCT. Leaf certificate is "
+              + sc.getLeafCertificate().getClass());
+    }
+    CertificateEntry ce =
+        CertificateEntry.createForX509Certificate((X509Certificate) sc.getLeafCertificate());
+
+    // a result is returned here, but I don't know what to do with it yet
```

Comment:
What's the options for statuses? I assume valid or not? If not valid, return an error?

Reply:
If not valid, it throws an exception, but I'm mucking around with internal APIs of this library. I'm trying to see if they will offer an API for this kind of stuff. Other options are:
https://github.com/google/conscrypt/blob/master/common/src/main/java/org/conscrypt/ct/VerifiedSCT.java#L25